packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -311,75 +409,140 @@ fn batches_from_shared_store(
+    unsafe extern "C" fn no_op(_: *mut std::ffi::c_void, _: usize, _: *mut std::ffi::c_void) {}
+
+    // https://github.com/denoland/rusty_v8/pull/926
+    //
+    // SAFETY: `mem.data` points to valid memory and is valid for `mem.size` bytes `no_op` will not
+    //         try to de-allocate share memory.
+    // TODO: Investigate, that this does not have any implications on reading/writing shared memory.
+    //       It's also not 100% clear what `ArrayBuffer` expects, is it ok to read/write while the
+    //       `ArrayBuffer` exists?)
+    let backing_store = unsafe {
+        v8::ArrayBuffer::new_backing_store_from_ptr(
+            mem.data.as_ptr().cast(),
+            mem.size,
+            no_op,
+            std::ptr::null_mut(),
+        )
+    };
+    let array_buffer = v8::ArrayBuffer::with_backing_store(scope, &backing_store.make_shared());
+
     let batch_id = mem.id();
-    mem_batch_to_js(mv8, batch_id, arraybuffer, persisted)
+    mem_batch_to_js(scope, batch_id, array_buffer.into(), persisted)
 }
 
-fn state_to_js<'m, 'a, 'b>(
-    mv8: &'m MiniV8,
+fn state_to_js<'s, 'a>(
+    scope: &mut v8::HandleScope<'s>,
     mut agent_batches: impl Iterator<Item = &'a AgentBatch>,
-    mut message_batches: impl Iterator<Item = &'b MessageBatch>,
-) -> Result<(mv8::Value<'m>, mv8::Value<'m>)> {
-    let js_agent_batches = mv8.create_array();
-    let js_message_batches = mv8.create_array();
+    mut message_batches: impl Iterator<Item = &'a MessageBatch>,
+) -> Result<(v8::Local<'s, v8::Value>, v8::Local<'s, v8::Value>)> {
+    // I'm not sure if we need to know the length beforehand or not
+    let js_agent_batches = v8::Array::new(scope, 0);
+    let js_message_batches = v8::Array::new(scope, 0);
 
     for (i_batch, (agent_batch, message_batch)) in agent_batches
         .by_ref()
         .zip(message_batches.by_ref())
         .enumerate()
     {
         let agent_batch = batch_to_js(
-            mv8,
+            scope,
             agent_batch.batch.segment().memory(),
             agent_batch.batch.segment().persisted_metaversion(),
         )?;
-        js_agent_batches.set(i_batch as u32, agent_batch)?;
+        js_agent_batches
+            .set_index(scope, i_batch as u32, agent_batch)
+            .ok_or_else(|| {
+                Error::V8(format!("Could not set index {i_batch} on js_agent_batches"))
+            })?;
 
         let message_batch = batch_to_js(
-            mv8,
+            scope,
             message_batch.batch.segment().memory(),
             message_batch.batch.segment().persisted_metaversion(),
         )?;
-        js_message_batches.set(i_batch as u32, message_batch)?;
+        js_message_batches
+            .set_index(scope, i_batch as u32, message_batch)
+            .ok_or_else(|| {
+                Error::V8(format!(
+                    "Could not set index {i_batch} on js_message_batches"
```

Comment:
Same as above, can we say what we're setting, we're not setting an index we're setting something _at_ an index. And can we describe the object properly, not just the variable name

Reply:
This was not addressed.

Reply:
Actually addressed in https://github.com/hashintel/hash/pull/414/commits/512bd6aafdb4475bccaacb0c61946dbf4980b70c